SensorNode/sensor_nodes/onBoxSensorNode.py

```diff
@@ -56,6 +57,26 @@ def __init__(self, config_path: str):
             # absolute() not necessary because of ASMSensorNode dir
             self.ff_log_dir = pathlib.Path(appdirs.user_log_dir('ASMSensorNode'), 'ffmpeg_logs')
 
+        self.extra_endpoints = sensor_params['extra_endpoints']
+        if self.extra_endpoints is not None:
+            if len(self.extra_endpoints) ==0:
+                self.extra_endpoints = None
+            else:
+                # Check if extra_endpoints a list 
+                try:
+                    assert isinstance(self.extra_endpoints, list)
+                except AssertionError as e:
+                    raise TypeError(f"Extra_endpoints is not None, expect type list, get{type(self.extra_endpoints)}") from None
+                # Check if url are properlt formatted
+                for url in self.extra_endpoints:
+                    parsed = parse.urlparse(url)
+                    if not (parsed.scheme and parsed.netloc):
+                        print(f"Detected Improper formatted URL: {url}")
```

Comment:
This is probably an error condition - this should result in some exception being thrown.  In general, internal code like this should not use print statements.